src/promptflow/promptflow/_sdk/_service/entry.py

```diff
@@ -53,25 +68,49 @@ def add_show_status_action(subparsers):
 
 def start_service(args):
     port = args.port
-    app, _ = create_app()
-    if port and is_port_in_use(port) and not args.force:
-        app.logger.warning(f"Service port {port} is used.")
-        raise UserErrorException(f"Service port {port} is used.")
-    if not port:
+    get_app()
+
+    def validate_port(port, force_start):
+        if is_port_in_use(port):
+            if force_start:
+                app.logger.warning(f"Force restart the service on the port {port}.")
+                kill_exist_service(port)
+            else:
+                app.logger.warning(f"Service port {port} is used.")
+                raise UserErrorException(f"Service port {port} is used.")
+
+    if port:
+        dump_port_to_config(port)
+        validate_port(port, args.force)
+    else:
         port = get_port_from_config(create_if_not_exists=True)
+        validate_port(port, args.force)
+    # Set host to localhost, only allow request from localhost.
+    cmd = [
+        sys.executable,
+        "-m",
+        "waitress",
+        "--host",
+        "127.0.0.1",
+        f"--port={port}",
+        "--call",
+        "promptflow._sdk._service.entry:get_app",
+    ]
+    if args.synchronous:
+        subprocess.call(cmd)
```

Comment:
shall we set the environment variable PF_NO_INTERACTIVE_LOGIN here?

Reply:
copy from service entry: https://github.com/microsoft/promptflow/blob/b74b863c1910bb5bc156fbc40018db12c2516838/src/promptflow/promptflow/_sdk/_service/entry.py#L86